core/biogeophys/cable_common.F90

```diff
@@ -684,6 +685,105 @@ subroutine get_unit(iunit)
 
   end subroutine get_unit
 
+  FUNCTION latitude_dimid(FileID)
+    !*## Purpose
+    !
+    ! Find the id of the latitude dimension
+    !
```

Comment:
Wanted to suggest an alternative design that would reduce repeats. You could read the dimensions in a functional manner across different variables. For example, the pseudocode would look like:

```fortran
SUBROUTINE read_dimid(FileID, possible_names)
    INTEGER :: FileID
    CHARACTER(LEN=*), INTENT(IN) :: possible_names
    ! Define other vars
    ! Run through the possible names until we find the correct one
    ! We take  possible_names[0] as default param name
    handle_error(ok, f"Failed to find any {possible_names(0)} from the list")
```
Then add [arrays of strings](https://fortran-lang.org/learn/quickstart/arrays_strings/#array-of-strings) as parameters. For example

```
  CHARACTER(LEN=10), DIMENSION(5), PARAMETER :: LatNames
  LatNames = ["y", "latitude", "lat", "lats", "Latitude"]
```

Reply:
I think I know what you mean- have the ```cable_common``` define ```LatNames, LonNames, TimeNames```, which can be imported by other modules as necessary and passed to the ```read_dimid``` function. I guess we don't lose any clarity either, because the passed variable name makes it clear which dimension we're retrieving.

Reply:
Hmm, maybe we could have these defined in a derived type called `PossibleNamesType` / similar, but we want these to be compile-time constants for efficiency as well. If you think it wouldn't make much of a difference (since it would probably be optimized in `Release` builds), then what do you think about going for derived type?

Reply:
I don't think something like that is necessary, I don't see any benefit to that over just having an array for each set of names. It lets us use ```PARAMETER``` as well.